core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java

```diff
@@ -63,6 +63,17 @@ public BuddyDTO addBuddyToAcceptingUser(UUID buddyUserID, String buddyNickName,
 		return BuddyDTO.createInstance(Buddy.getRepository().save(buddy));
 	}
 
+	public void removeBuddyAfterConnectRejection(UUID idOfRequestingUser, UUID buddyID)
+	{
+		User user = User.getRepository().findOne(idOfRequestingUser);
+		Buddy buddy = getEntityByID(buddyID);
+
+		if (buddy != null)
+		{
+			user.removeBuddy(buddy);
```

Comment:
The removeBuddy method saves the user record, so the buddy get removed that way. The saveBuddy also saves the record, so I thought it is consistent that way.